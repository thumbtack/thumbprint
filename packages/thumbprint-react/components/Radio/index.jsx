import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import * as tokens from '@thumbtack/thumbprint-tokens';
import warning from 'warning';
import styles from './index.module.scss';

const borderColor = {
    disabled: tokens.tpColorGray300,
    checked: tokens.tpColorBlue,
    unchecked: tokens.tpColorGray,
    error: tokens.tpColorRed,
};

const labelCursor = {
    disabled: 'default',
    checked: 'pointer',
    unchecked: 'pointer',
    error: 'pointer',
};

const backgroundColor = {
    disabled: tokens.tpColorGray200,
    checked: tokens.tpColorWhite,
    unchecked: tokens.tpColorWhite,
    error: tokens.tpColorWhite,
};

/**
 * Color of the inner circle when the radio is checked.
 */
const circleColor = {
    disabled: tokens.tpColorGray,
    checked: tokens.tpColorBlue,
    error: tokens.tpColorRed,
};

const textColor = {
    disabled: tokens.tpColorGray,
    checked: 'inherit',
    unchecked: 'inherit',
    error: tokens.tpColorRed,
};

const getUIState = ({ isChecked, isDisabled, hasError }) => {
    if (isDisabled) {
        return 'disabled';
    }

    if (hasError) {
        return 'error';
    }

    if (isChecked) {
        return 'checked';
    }

    return 'unchecked';
};

const Radio = ({
    children,
    dataTest,
    hasError,
    id,
    isChecked,
    isDisabled,
    labelPadding,
    name,
    onChange,
    radioVerticalAlign,
}) => {
    const uiState = getUIState({ isChecked, isDisabled, hasError });

    warning(
        children || id,
        'If `children` is not provided to `Radio`, the `id` prop must be used to link the `Radio` to a `label` element.',
    );

    return (
        // eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for
        <label
            className={classNames(styles.root, {
                [styles.rootRadioVerticalAlignTop]: radioVerticalAlign === 'top',
                [styles.rootRadioVerticalAlignCenter]: radioVerticalAlign === 'center',
            })}
            style={{ padding: labelPadding, cursor: labelCursor[uiState] }}
        >
            <input
                className={styles.input}
                type="radio"
                id={id}
                onChange={e => onChange(e.target.checked, id)}
                checked={isChecked}
                name={name}
                disabled={isDisabled}
                data-test={dataTest}
            />

            <svg
                className={styles.radioImage}
                width="20"
                height="20"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g fill={backgroundColor[uiState]} fillRule="evenodd">
                    <circle stroke={borderColor[uiState]} strokeWidth="2" cx="10" cy="10" r="9" />

                    {isChecked && <circle fill={circleColor[uiState]} cx="10" cy="10" r="6" />}
                </g>
            </svg>

            {children && (
                <span className={styles.text} style={{ color: textColor[uiState] }}>
                    {children}
                </span>
            )}
        </label>
    );
};

Radio.propTypes = {
    /**
     * Disable the input and the label.
     */
    isDisabled: PropTypes.bool,
    /**
     * Text or elements that appear within the label. If `children` is not provided, the developer
     * must use the `Radio`'s `id` prop to associate it with a custom `<label>` element.
     */
    children: PropTypes.node,
    /**
     * The `id` is added to the radio button as an HTML attribute and passed to the `onChange`
     * function.
     */
    id: PropTypes.string,
    /**
     * Boolean that determines if the radio is checked.
     */
    isChecked: PropTypes.bool,
    /**
     * Makes the radio and text color red.
     */
    hasError: PropTypes.bool,
    /**
     * Radio buttons that have the same name attribute are in the same radio button group. Only one
     * radio in a group can be selected at a time. All of the radio buttons in the group must share
     * a value that is unique to the page. This is required for keyboard navigation.
     */
    name: PropTypes.string.isRequired,
    /**
     * Determine that padding that gets applied to the label. This can be used
     * to increase the labelâ€™s click target. The value supplied should be a
     * string with a unit such as `8px` or `8px 16px`.
     */
    labelPadding: PropTypes.string,
    /**
     * Function that runs when a new radio button is selected. It receives the new boolean value
     * and the provided `id` as such: `props.onChange(e.target.checked, props.id)`.
     */
    onChange: PropTypes.func.isRequired,
    /**
     * A selector hook into the React component for use in automated testing environments. It is
     * applied internally to the `<input />` element.
     */
    dataTest: PropTypes.string,
    /**
     * Determines how the radio button input will be vertically aligned relative to `props.children`.
     */
    radioVerticalAlign: PropTypes.oneOf(['top', 'center']),
};

Radio.defaultProps = {
    children: null,
    id: null,
    isDisabled: false,
    isChecked: false,
    hasError: false,
    labelPadding: '14px 0',
    dataTest: undefined,
    radioVerticalAlign: 'center',
};

export default Radio;
