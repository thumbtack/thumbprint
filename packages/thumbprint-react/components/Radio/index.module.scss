@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';

.root {
    position: relative;
    display: flex;

    &RadioVerticalAlignTop {
        align-items: flex-start;
    }

    &RadioVerticalAlignCenter {
        align-items: center;
    }
}

.input {
    position: absolute;
    opacity: 0;
    z-index: -1;
    width: 1px;
    height: 1px;

    &:focus ~ .radioImage {
        box-shadow: 0 0 0px 4px $tp-color__gray-300;
    }
}

.radioImage {
    width: 20px;
    height: 20px;
    border-radius: $tp-border-radius__full;

    // Margin that makes the input appear decently aligned against
    // various Thumbprint <Text> sizes
    .rootRadioVerticalAlignTop & {
        margin-top: 1px;
    }

    .rootRadioVerticalAlignCenter & {
        margin-top: -1px;
    }
}

.text {
    // This must take up the remaining available space for content that must be right-aligned.
    flex: 1 0 0%;
    // 'min-width' is needed to allow text within the radio to truncate.
    min-width: 0;
    font-size: $tp-font__body__1__size;
    font-weight: 400;
    padding-left: 8px;
}
