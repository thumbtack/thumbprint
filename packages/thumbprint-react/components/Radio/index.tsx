import React from 'react';
import classNames from 'classnames';
import * as tokens from '@thumbtack/thumbprint-tokens';
import warning from 'warning';
import styles from './index.module.scss';

const borderColor = {
    disabled: tokens.tpColorGray300,
    checked: tokens.tpColorBlue,
    unchecked: tokens.tpColorGray,
    error: tokens.tpColorRed,
};

const labelCursor = {
    disabled: 'default',
    checked: 'pointer',
    unchecked: 'pointer',
    error: 'pointer',
};

const backgroundColor = {
    disabled: tokens.tpColorGray200,
    checked: tokens.tpColorWhite,
    unchecked: tokens.tpColorWhite,
    error: tokens.tpColorWhite,
};

/**
 * Color of the inner circle when the radio is checked.
 */
const circleColor = {
    disabled: tokens.tpColorGray,
    checked: tokens.tpColorBlue,
    unchecked: undefined,
    error: tokens.tpColorRed,
};

const textColor = {
    disabled: tokens.tpColorGray,
    checked: 'inherit',
    unchecked: 'inherit',
    error: tokens.tpColorRed,
};

type UiState = 'disabled' | 'error' | 'checked' | 'unchecked';

const getUIState = ({
    isChecked,
    isDisabled,
    hasError,
}: Pick<PropTypes, 'isDisabled' | 'hasError' | 'isChecked'>): UiState => {
    if (isDisabled) {
        return 'disabled';
    }

    if (hasError) {
        return 'error';
    }

    if (isChecked) {
        return 'checked';
    }

    return 'unchecked';
};

interface PropTypes {
    /**
     * Disable the input and the label.
     */
    isDisabled?: boolean;
    /**
     * Text or elements that appear within the label. If `children` is not provided, the developer
     * must use the `Radio`'s `id` prop to associate it with a custom `<label>` element.
     */
    children?: React.ReactNode;
    /**
     * The `id` is added to the radio button as an HTML attribute and passed to the `onChange`
     * function.
     */
    id?: string;
    /**
     * Boolean that determines if the radio is checked.
     */
    isChecked?: boolean;
    /**
     * Adds the `required` HTML attribute.
     */
    isRequired?: boolean;
    /**
     * Makes the radio and text color red.
     */
    hasError?: boolean;
    /**
     * Radio buttons that have the same name attribute are in the same radio button group. Only one
     * radio in a group can be selected at a time. All of the radio buttons in the group must share
     * a value that is unique to the page. This is required for keyboard navigation.
     */
    name: string;
    /**
     * Determine that padding that gets applied to the label. This can be used
     * to increase the labelâ€™s click target. The value supplied should be a
     * string with a unit such as `8px` or `8px 16px`.
     */
    labelPadding?: string;
    /**
     * Function that runs when a new radio button is selected. It receives the new boolean value
     * and the provided `id` as such: `props.onChange(e.target.checked, props.id)`.
     */
    onChange: (isChecked: boolean, id?: string) => void;
    /**
     * A selector hook into the React component for use in automated testing environments. It is
     * applied internally to the `<input />` element.
     */
    dataTest?: string;
    /**
     * Determines how the radio button input will be vertically aligned relative to `props.children`.
     */
    radioVerticalAlign?: 'top' | 'center';
}

export default function Radio({
    children = null,
    dataTest,
    id,
    isChecked = false,
    isDisabled = false,
    isRequired = false,
    hasError = false,
    labelPadding = '14px 0',
    name,
    onChange,
    radioVerticalAlign = 'center',
}: PropTypes): JSX.Element {
    const uiState = getUIState({ isChecked, isDisabled, hasError });

    warning(
        children || id,
        'If `children` is not provided to `Radio`, the `id` prop must be used to link the `Radio` to a `label` element.',
    );

    return (
        // eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for
        <label
            className={classNames(styles.root, {
                [styles.rootRadioVerticalAlignTop]: radioVerticalAlign === 'top',
                [styles.rootRadioVerticalAlignCenter]: radioVerticalAlign === 'center',
            })}
            style={{ padding: labelPadding, cursor: labelCursor[uiState] }}
        >
            <input
                className={styles.input}
                type="radio"
                id={id}
                onChange={(event): void => onChange(event.target.checked, id)}
                checked={isChecked}
                name={name}
                disabled={isDisabled}
                data-test={dataTest}
                required={isRequired}
            />

            <svg
                className={styles.radioImage}
                width="20"
                height="20"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g fill={backgroundColor[uiState]} fillRule="evenodd">
                    <circle stroke={borderColor[uiState]} strokeWidth="2" cx="10" cy="10" r="9" />

                    {isChecked && <circle fill={circleColor[uiState]} cx="10" cy="10" r="6" />}
                </g>
            </svg>

            {children && (
                <span className={styles.text} style={{ color: textColor[uiState] }}>
                    {children}
                </span>
            )}
        </label>
    );
}
