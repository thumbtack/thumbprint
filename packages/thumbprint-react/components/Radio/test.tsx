import React from 'react';
import { mount } from 'enzyme';
import Radio from './index';

test('renders radio input', () => {
    const wrapper = mount(
        <Radio name="duck" onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.find('input').prop('type')).toBe('radio');
    expect(wrapper).toMatchSnapshot();
});

test('renders `children` passed in', () => {
    const wrapper = mount(
        <Radio name="duck" onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.text()).toBe('Goose');
    expect(wrapper).toMatchSnapshot();
});

test('if no `children` are passed in', () => {
    const wrapper = mount(<Radio name="duck" onChange={jest.fn()} id="no-children" />);
    expect(wrapper).toMatchSnapshot();
});

test('renders `hasError` state', () => {
    const wrapperA = mount(
        <Radio name="duck" hasError onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    const wrapperB = mount(
        <Radio name="duck" isChecked hasError onChange={jest.fn()}>
            Goose
        </Radio>,
    );

    expect(wrapperA).toMatchSnapshot();
    expect(wrapperB).toMatchSnapshot();
});

test('renders `input` within `label`', () => {
    const wrapper = mount(
        <Radio name="duck" onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.find('label input').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
});

test('adds `disabled` attribute', () => {
    const wrapper = mount(
        <Radio name="duck" isDisabled onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.find('input').prop('disabled')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
});

test('is not checked by default', () => {
    const wrapper = mount(
        <Radio name="duck" onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.find('input').prop('checked')).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
});

test('is checked when `isChecked` is set to true', () => {
    const wrapper = mount(
        <Radio name="duck" isChecked onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.find('input').prop('checked')).toBe(true);
    expect(wrapper).toMatchSnapshot();
});

test('applies `labelPadding` as padding on the label', () => {
    const wrapper = mount(
        <Radio name="duck" labelPadding="10px 20px" onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect((wrapper.find('label').prop('style') || {}).padding).toBe('10px 20px');
    expect(wrapper).toMatchSnapshot();
});

test('applies className corresponding to `radioVerticalAlign` on the label', () => {
    const wrapper = mount(
        <Radio name="duck" radioVerticalAlign="top" onChange={jest.fn()}>
            Goose
        </Radio>,
    );

    expect(wrapper.find('label').hasClass('rootRadioVerticalAlignTop'));
    expect(wrapper).toMatchSnapshot();
});

test('calls `onChange` function when supplied', () => {
    const onChange = jest.fn();

    const wrapper = mount(
        <Radio name="duck" onChange={onChange}>
            Goose
        </Radio>,
    );

    wrapper.find('input').simulate('change', { target: { checked: true } });
    expect(onChange).toHaveBeenCalledTimes(1);

    wrapper.find('input').simulate('change', { target: { checked: true } });
    expect(onChange).toHaveBeenCalledTimes(2);
});

test('passes correct new value and `id` to `onChange` function', () => {
    const onChange = jest.fn();

    const wrapper = mount(
        <Radio name="duck" id="goose" onChange={onChange} isChecked>
            Goose
        </Radio>,
    );

    wrapper.find('input').simulate('change', { target: { checked: false } });
    expect(onChange).toHaveBeenCalledWith(false, 'goose');

    wrapper.find('input').simulate('change', { target: { checked: true } });
    expect(onChange).toHaveBeenCalledWith(true, 'goose');
});

test('adds `dataTest` prop', () => {
    const wrapper = mount(
        <Radio name="duck" dataTest="Duck" onChange={jest.fn()}>
            Goose
        </Radio>,
    );
    expect(wrapper.find('input').prop('data-test')).toEqual('Duck');
    expect(wrapper).toMatchSnapshot();
});
