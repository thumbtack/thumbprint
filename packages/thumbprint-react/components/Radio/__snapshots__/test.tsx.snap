// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds \`dataTest\` prop 1`] = `
<Radio
  dataTest="Duck"
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      data-test="Duck"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`adds \`disabled\` attribute 1`] = `
<Radio
  isDisabled={true}
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "default",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={true}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#fafafa"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#e9eced"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "#d3d4d5",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`applies \`labelPadding\` as padding on the label 1`] = `
<Radio
  labelPadding="10px 20px"
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "10px 20px",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`applies className corresponding to \`radioVerticalAlign\` on the label 1`] = `
<Radio
  name="duck"
  onChange={[MockFunction]}
  radioVerticalAlign="top"
>
  <label
    className="root rootRadioVerticalAlignTop"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`if no \`children\` are passed in 1`] = `
<Radio
  id="no-children"
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      id="no-children"
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
  </label>
</Radio>
`;

exports[`is checked when \`isChecked\` is set to true 1`] = `
<Radio
  isChecked={true}
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={true}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#009fd9"
          strokeWidth="2"
        />
        <circle
          cx="10"
          cy="10"
          fill="#009fd9"
          r="6"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`is not checked by default 1`] = `
<Radio
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`renders \`children\` passed in 1`] = `
<Radio
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`renders \`hasError\` state 1`] = `
<Radio
  hasError={true}
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#ff5a5f"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "#ff5a5f",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`renders \`hasError\` state 2`] = `
<Radio
  hasError={true}
  isChecked={true}
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={true}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#ff5a5f"
          strokeWidth="2"
        />
        <circle
          cx="10"
          cy="10"
          fill="#ff5a5f"
          r="6"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "#ff5a5f",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`renders \`input\` within \`label\` 1`] = `
<Radio
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;

exports[`renders radio input 1`] = `
<Radio
  name="duck"
  onChange={[MockFunction]}
>
  <label
    className="root rootRadioVerticalAlignCenter"
    style={
      Object {
        "cursor": "pointer",
        "padding": "14px 0",
      }
    }
  >
    <input
      checked={false}
      className="input"
      disabled={false}
      name="duck"
      onChange={[Function]}
      required={false}
      type="radio"
    />
    <svg
      className="radioImage"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        fillRule="evenodd"
      >
        <circle
          cx="10"
          cy="10"
          r="9"
          stroke="#d3d4d5"
          strokeWidth="2"
        />
      </g>
    </svg>
    <span
      className="text"
      style={
        Object {
          "color": "inherit",
        }
      }
    >
      Goose
    </span>
  </label>
</Radio>
`;
